### DISCUSSION TITLE
LSTNet Multivariate Dataset to Predict One Target

### DISCUSSION THREAD

**Discussion by jonbeckman** [awslabs/gluonts#1388] [Q&A]
I am looking to implement a LSTNet model that uses a multivariate dataset to predict one target (N:1). 

For example, I have created a DeepAR model that works okay with the following dataset:
```
dataset = ListDataset(
    [{
        FieldName.START: df.index[0],
        FieldName.TARGET: df['target'][:],
        FieldName.FEAT_DYNAMIC_REAL: [df['feature_a'][:], df['feature_b'][:], df['feature_c'][:]],
    }],
    freq=config.DATASET_FREQ
)
```
In this example, it is using multiple time-series to help predict future target values (N:1). 

However, I am experimenting with benchmarking other models to determine if they perform better. In this case, I am interested in using the same dataset from the aforementioned example in LSTNet. Though, I can only seem to get the LSTNet model to run when I have the dataset configured to predict N:N targets. That is I provide the target and the three feature time-series which LSTNet then attempts to predict all four. For example:
```
dataset = ListDataset(
    [{
        FieldName.START: df.index[0],
        FieldName.TARGET: [df['target'][:], df['feature_a'][:], df['feature_b'][:], df['feature_c'][:]],
    }],
    freq=config.DATASET_FREQ,
    one_dim_target=False
)
```

I have tried so many different combinations of the dataset, multivariate grouper, etc to no avail. Can someone please point me in the right direction with this problem? Let me know if you have any questions or need more information.

**Comment by pratikgehlott** [comment]
hey, can you share your full code of DeepAR with the multivariate dataset?
